fix(mcp): handle CancelledError in MCPSessionManager.create_session

raben opened the pull request:
### Link to Issue or Description of Change

**1. Link to an existing issue:**

- Closes: #3708

**2. Description:**

**Problem:**
When an MCP server returns an HTTP error (e.g., 401, 403), the MCP SDK uses anyio cancel scopes internally, which raise `CancelledError`. Since `CancelledError` is a `BaseException` (not `Exception`) in Python 3.8+, the existing `except Exception` block in `MCPSessionManager.create_session()` does not catch it.

This causes the error to propagate uncaught, leading to:
- Application hangs
- "ASGI callable returned without completing response" errors in web frameworks
- Inability to handle MCP connection failures gracefully

**Solution:**
This fix explicitly catches `asyncio.CancelledError` before the general `Exception` handler and converts it to a `ConnectionError` with a descriptive message. This allows:
- Proper error handling and cleanup
- Exit stack cleanup to prevent resource leaks
- Graceful error propagation that applications can handle

### Testing Plan

**Unit Tests:**

- [x] I have added or updated unit tests for my change.
- [x] All unit tests pass locally.

```
tests/unittests/tools/mcp_tool/test_mcp_session_manager.py::TestMCPSessionManager::test_create_session_cancelled_error PASSED
```

**Manual End-to-End (E2E) Tests:**

Tested with an MCP server returning HTTP 403 Forbidden:
1. Before fix: Application hangs with "ASGI callable returned without completing response"
2. After fix: `ConnectionError` is raised with descriptive message, allowing proper error handling

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/google/adk-python/blob/main/CONTRIBUTING.md) document.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have manually tested my changes end-to-end.
- [x] Any dependent changes have been merged and published in downstream modules.

### Additional context

This issue affects multiple users as described in #3708. The root cause is that `asyncio.CancelledError` inherits from `BaseException` (not `Exception`) in Python 3.8+, so it escapes the existing exception handler.

raben committed:
fix(mcp): handle CancelledError in MCPSessionManager.create_session

When an MCP server returns an HTTP error (e.g., 401, 403), the MCP SDK
uses anyio cancel scopes internally, which raise CancelledError. Since
CancelledError is a BaseException (not Exception) in Python 3.8+, the
existing `except Exception` block does not catch it.

This causes the error to propagate uncaught, leading to:
- Application hangs
- ASGI callable returned without completing response errors
- Inability to handle MCP connection failures gracefully

This fix explicitly catches asyncio.CancelledError and converts it to
a ConnectionError with a descriptive message, allowing proper error
handling and cleanup.

Fixes #3708

ryanaiagent commented:
Hi @raben , Thank you for your contribution! We appreciate you taking the time to submit this pull request. Can you fix the failing lint error. You can run autoformat.sh. 
Let us know once you are done so that we can continue with the review

raben committed:
style(mcp): apply autoformat to fix lint errors

- Format mcp_session_manager.py (remove unnecessary f-string prefix)
- Format test_mcp_session_manager.py (consolidate pytest.raises call)

shovelmn12 commented:
I am still getting "Warning: Error during MCP session cleanup for session_9ae81d018baca2ff4b59e7785d1e3f08: Attempted to exit a cancel scope that isn't the current tasks's current cancel scope"

ryanaiagent commented:
Hi @wukath , can you please review this.

haranrk commented:
Thanks for your contribution, @raben! This has been fixed in cce430da, so I'm closing this PR.
